Signal/src/AppDelegate.m

```diff
@@ -48,7 +49,7 @@
 static NSString *const kURLSchemeSGNLKey                = @"sgnl";
 static NSString *const kURLHostVerifyPrefix             = @"verify";
 
-@interface AppDelegate ()
+@interface AppDelegate () <AppContext>
```

Comment:
It might make more sense to have separate `MainAppContext` and `AppExtensionContext` classes rather than a protocol implemented by the AppDelegate, etc.  Hopefully the contexts are stateless, and this might help avoid bad references with the odd lifecycle around app extensions.